src/python/WMCore/Services/DBS/DBSErrors.py

```diff
@@ -58,3 +63,51 @@ class DBSReaderError(DataMgmtError):
     """
     def __init__(self, msg, **data):
         DataMgmtError.__init__(self, msg, 1002, **data)
+
+
+class DBSError():
+    """
+    DBSError provides generic interface for DBS (Go-based) errors
+    """
+    def __init__(self, ex):
+        try:
+            if hasattr(ex, "body"):
+                # case of DBSClient HTTPError
+                self.data = json.loads(ex.body)[0]
+            else:
+                self.data = json.loads(ex)[0]
+        except Exception as exp:
+            self.data = str(ex)
+
+    def getHttpCode(self):
+        """
+        :return: HTTP error code
+        """
+        if isinstance(self.data, dict):
+            return self.data['http']['code']
+        return self.data
+
+    def getServerCode(self):
+        """
+        :return: DBS server error code which is defined here
+-        https://github.com/dmwm/dbs2go/blob/master/dbs/errors.go
+        """
+        if isinstance(self.data, dict):
+            return self.data['error']['code']
+        return self.data
```

Comment:
Same comment here.

Reply:
return type should be consistent, and I'll return it 0.